sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/PropertiesAccessCorrectField.cs

```diff
@@ -49,27 +50,41 @@ protected override IEnumerable<FieldData> FindFieldAssignments(IPropertySymbol p
 
             // we only keep information for the first location of the symbol
             var assignments = new Dictionary<IFieldSymbol, FieldData>();
-            foreach (var node in setter.DescendantNodes())
+            FillAssignments(setter, true);
+
+            // If there're no candidate variables, we'll try to inspect one local method invocation with value as argument
+            if (assignments.Count == 0
+                && (setter.ExpressionBody()?.Expression ?? SingleInvocation(setter.Body)) is ExpressionSyntax expression
+                && FindInvokedMethod(compilation, property.ContainingType, expression) is MethodDeclarationSyntax invokedMethod)
             {
-                FieldData? foundField = null;
-                if (node is AssignmentExpressionSyntax assignment && node.IsKind(SyntaxKind.SimpleAssignmentExpression))
-                {
-                    foundField = assignment.Left.DescendantNodesAndSelf().OfType<ExpressionSyntax>()
-                        .Select(x => ExtractFieldFromExpression(AccessorKind.Setter, x, compilation))
-                        .FirstOrDefault(x => x != null);
-                }
-                else if (node is ArgumentSyntax argument && IsRefOrOut(argument.RefOrOutKeyword))
-                {
-                    foundField = ExtractFieldFromExpression(AccessorKind.Setter, argument.Expression, compilation);
-                }
-                if (foundField.HasValue && !assignments.ContainsKey(foundField.Value.Field))
-                {
-                    assignments.Add(foundField.Value.Field, foundField.Value);
-                }
+                FillAssignments(invokedMethod, false);
             }
+
             return assignments.Values;
 
             bool IsRefOrOut(SyntaxToken node) => node.IsKind(SyntaxKind.RefKeyword) || node.IsKind(SyntaxKind.OutKeyword);
+
+            void FillAssignments(SyntaxNode root, bool useFieldLocation)
```

Comment:
I think it would be better to extract this outside of this method. The method is big enough already and like this it hides the dependencies (compilation, assignments, IsRefOrOut)